srcjs/init_shiny.js

```diff
@@ -288,6 +288,53 @@ function initShiny() {
       initialValues['.clientdata_output_' + id + '_height'] = this.offsetHeight;
     }
   });
+
+  function getComputedBgColor(el) {
+    if (!el) {
+      // Top of document, can't recurse further
+      return null;
+    }
+
+    let props = window.getComputedStyle(el);
+    let bgColor = props.getPropertyValue("background-color");
+    let m = bgColor.match(/^rgba\(\s*([\d.]+)\s*,\s*([\d.]+)\s*,\s*([\d.]+)\s*,\s*([\d.]+)\s*\)$/);
+    if (bgColor === "transparent" || (m && parseFloat(m[4]) === 0)) {
+      // No background color on this element. See if it has a background image.
+      let bgImage = props.getPropertyValue("background-image");
+      if (bgImage && bgImage !== "none") {
+        // Failed to detect background color, since it has a background image
+        return null;
+      } else {
+        // Recurse
+        return getComputedBgColor(el.parentElement);
+      }
+    }
+    return bgColor;
+  }
+
+  // Compute the color property of an a tag, scoped within the element
+  function getComputedLinkColor(el) {
+    let a = document.createElement("a");
+    a.href = "/";
+    let div = document.createElement("div");
+    div.style.position = "absolute";
+    div.style.top = "-1000px";
```

Comment:
I would set `left`, `width`, and `height` as well, just so there's no question about how it will lay out.

Reply:
Actually to be really safe you could make the inline CSS properties `!important`; you have to use this syntax (TIL): `div.style.setProperty("position", "absolute", "important");`